### ISSUE TITLE
How to properly fix random seed with TFT?

### ISSUE THREAD

**Issue by monkiravn** [jdb78/pytorch-forecasting#401]
### Expected behavior

Hello guys
I wonder how to fix seed to get reproducibility of my experiments

Right now I'm using this function before the start of the training
`pl.seed_everything(42)`

### Actual behavior

However, it doesn't work. It is caused by dropout because when I set dropout = 0 it had worked.

```
# configure network and trainer
pl.seed_everything(42)
trainer = pl.Trainer(
    deterministic=True,
    gpus=[0],
    gradient_clip_val=0.1,
)


tft = TemporalFusionTransformer.from_dataset(
    training,
    learning_rate=0.03,
    hidden_size=128,
    attention_head_size=2,
    dropout=0.3,  
    hidden_continuous_size=32,  
    output_size=7,  
    loss=QuantileLoss(),
    reduce_on_plateau_patience=4,
)

trainer.fit(
    tft,
    train_dataloader=train_dataloader,
    val_dataloaders=val_dataloader,
)
```

So, how do I do to fix this issue?

Thanks,

[This is my colab notebook.](https://colab.research.google.com/drive/1KYJJ9cwPDE_THZO63If0BLuWjlZv4AnL?usp=sharing)

**Comment by jdb78** [comment]
I believe the issue is that need to create a new dataloader from the dataset after setting the seed for the second training.

**Comment by monkiravn** [comment]
> I believe the issue is that need to create a new dataloader from the dataset after setting the seed for the second training.

Hi Jan Beitner,
I have tried but it doesn't work.
I also have tried in example tutorial (stallion). In 3 times of training model, the prediction results are different.
```
Frist: tensor(305.4471)

Second: tensor(319.3336)

Third: tensor(287.0847)
```
You can see this colab notebook[ here](https://drive.google.com/file/d/1Ia0PkaI45bdocfvmvsShY2CCdLYyfOe9/view?usp=sharing)

**Comment by LumingSun** [comment]
What about adding `shuffle=False` in `to_dataloader()`?

**Comment by monkiravn** [comment]
> What about adding `shuffle=False` in `to_dataloader()`?

Hi Luming Sun,
Thanks for your suggestion, I have tried but it hasn't worked too.
